File: 2163-kth-distinct-string-in-an-array/2163-kth-distinct-string-in-an-array.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        HashMap <String , Integer > hs = new HashMap <>();
        for (String a : arr) {
            hs.put(a,hs.getOrDefault(a,0)+1);
        }
        for (String v : arr) {
            if (hs.get(v) == 1) {
                --k;
                if (k == 0) {
                    return v;
                }
            }
        }
        return "";
    }
}

File: 2163-kth-distinct-string-in-an-array/2163-kth-distinct-string-in-an-array.cpp
#include <unordered_map>
#include <string>
#include <vector>

class Solution {
public:
    std::string kthDistinct(std::vector<std::string>& arr, int k) {
        std::unordered_map<std::string, int> hs;
        for (const std::string& a : arr) {
            hs[a]++;
        }
        for (const std::string& v : arr) {
            if (hs[v] == 1) {
                --k;
                if (k == 0) {
                    return v;
                }
            }
        }
        return "";
    }
};

